src/server/game/Conditions/ConditionMgr.cpp

```diff
@@ -2302,11 +2302,6 @@ bool ConditionMgr::isConditionTypeValid(Condition* cond)
         break;
     case CONDITION_WORLD_STATE:
     {
-        if (!sWorld->getWorldState(cond->ConditionValue1))
-        {
-            LOG_ERROR("sql.sql", "World state condition has non existing world state in value1 ({}), skipped", cond->ConditionValue1);
-            return false;
-        }
```

Comment:
why

Reply:
This check does prevent 0 being used as valid world state value. Working with 0/1 values is handy, since it can be used directly with if like `if (sWorld->getWorldState(STV_FISHING_ANNOUNCE_POOLS_DESPAN)) {...}`
Also this is the first usage of world states at condition db, so I guess allowing 0 should not affect other places.

Reply:
Cool